VideoPress: hide VideoPress block and update Video block description

thedebian opened the pull request:
Fixes Automattic/greenhouse#990
Fixes #11480

#### Changes proposed in this Pull Request:

The VideoPress embed block is confusing when using VideoPress.
This PR hides the VideoPress embed block in favor of the Video block (where we could still embed a VideoPress URL there).
It also updates the video block example and description to be more VideoPress-specific when VideoPress is enabled.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* On a Jetpack site with VideoPress enabled
* Create a new post or edit an existing one
* Type /videopress
* ✅ The video block should appear and the VideoPress embed block should not be visible

![Capture d’écran 2021-11-15 à 12 06 41](https://user-images.githubusercontent.com/1420594/141771622-10a07470-838b-4469-bad1-337bb1cfd302.png)

* Clear your line and click on the "+" button 
* Search for "videopress"
* ✅ The video block should appear and the VideoPress embed block should not be visible

![Capture d’écran 2021-11-15 à 12 07 39](https://user-images.githubusercontent.com/1420594/141771788-c51ccec8-18ff-4758-a8e1-dc51adc421cb.png)

* Click on "Browse all"
* ✅ The VideoPress embed block should not be visible in this list
* ✅ The Video block should have an updated example and description

![Capture d’écran 2021-11-15 à 12 09 20](https://user-images.githubusercontent.com/1420594/141771981-e18c289e-154b-45dd-9bc7-f1ce1be0be01.png)

* Add a video block
* Click on "Insert from URL"
* Insert a VideoPress link ( you can try https://videopress.com/v/Ygmx4akX )
* ✅ The block should be transformed to a VideoPress embed

![Capture d’écran 2021-11-15 à 12 11 03](https://user-images.githubusercontent.com/1420594/141772205-5054ac0f-97a9-4a44-9087-d50b4a262959.png)

* In a new tab, go to your Jetpack settings and disable VideoPress
* Refresh your edited post 
* ✅ The VideoPress embed block should be available again and the video block description should be the default core video description (no mentions of VideoPress)

thedebian committed:
[not verified] Hide core/embed/videopress block from the editor

thedebian committed:
[not verified] Update video block description when using VideoPress

thedebian committed:
[not verified] Update video block example with videopress-specific content when VideoPress is available, return an image instead of a video block when example is requested

roundhill reviewed (changes requested):
Tested well for me, have one suggestion for the image url.

thedebian commented on projects/plugins/jetpack/extensions/blocks/videopress/editor.js:
```diff
@@ -113,6 +133,18 @@ const addVideoPressSupport = ( settings, name ) => {
+			caption: __( 'Upload or embed videos with VideoPress', 'jetpack' ),
+			isVideoPressExample: true,
+			src:
+				'https://s0.wp.com/wp-content/themes/a8c/jetpackme-new/images/2021/videopress/device-mashup-desktop.jpg',
```
That's perfect, thanks !

thedebian committed:
[not verified] Update example image URL for video block

thedebian committed:
[not verified] Make sure video block and core/embed are updated only when VideoPress is available

thedebian commented:
@roundhill I updated the image URL and also added some check to make sure we only hide the core/embed VideoPress block and update the core/video example when VideoPress is available (the testing instructions have been updated too)

roundhill reviewed (dismissed):
Latest changes look good, looks it needs a rebase 😬

roundhill reviewed (dismissed):
Still good after rebase 👍

jeherve commented on projects/plugins/jetpack/extensions/blocks/videopress/editor.js:
```diff
@@ -114,6 +138,17 @@ const addVideoPressSupport = ( settings, name ) => {
+		settings.example.attributes = {
+			caption: __( 'Upload or embed videos with VideoPress', 'jetpack' ),
+			isVideoPressExample: true,
+			src: 'https://videopress2.files.wordpress.com/2021/11/videopress-devices.jpg',
```
In the past we've uploaded them to the plugin, so they can be served locally. Here is an example:
https://github.com/Automattic/jetpack/blob/2704efac24c831348233c68f6ab35893d706ddf7/projects/plugins/jetpack/extensions/blocks/tiled-gallery/index.js#L32-L40

Should we do that here too?

thedebian replied:
Sure, I'll update the PR ! :)

jeherve commented on projects/plugins/jetpack/extensions/blocks/videopress/editor.js:
```diff
@@ -114,6 +138,17 @@ const addVideoPressSupport = ( settings, name ) => {
+			'jetpack'
+		);
+		settings.example.attributes = {
+			caption: __( 'Upload or embed videos with VideoPress', 'jetpack' ),
```
Since we have a description right below the example, do we need the caption? Maybe it would be enough if the image showed a Play button on top of a blurry image, so folks understand it's a player?

thedebian replied:
The idea was to emphasize on "VideoPress" so people can know we're actually doing something and it's the right block to use, but this part is so hidden that I don't really know what would be best tbh; I'm okay with removing the caption though, it doesn't add much information while requiring a new translation, so I'll update the PR 👍

jeherve reviewed (commented):
It all works well. I only have a couple of questions about the example.

**********

> Click on "Insert from URL"
> Insert a VideoPress link ( you can try videopress.com/v/Ygmx4akX )
> ✅ The block should be transformed to a VideoPress embed

This works well. I was wondering if in the future, we could do a full transform so you get Video settings in the editor sidebar?

thedebian committed:
[not verified] VideoPress: remove caption from example and store example image locally

thedebian commented:
Last changes : rebase, remove caption and store example image locally

jeherve reviewed (commented):
The thumbnail seems to be cut off for me:
![image](https://user-images.githubusercontent.com/426388/145261166-6f0217a1-6e24-414b-8b4e-2e03ae14afe0.png)

Is that to be expected?

(this site is not using the Gutenberg plugin)

thedebian commented:
> Is that to be expected?

🤦 Of course not, I must have had some client cache or whatever when testing, I'm really sorry for the useless roundtrip, I'll update that !

jeherve reviewed (dismissed):
I cannot reproduce the above issue anymore. This should be good to merge now, after a `git merge origin/master`

samiff commented:
@thedebian Hmm I'm seeing the block preview with scrollbars 🤔 

![Screen Shot 2021-12-16 at 11 40 49 AM](https://user-images.githubusercontent.com/15803018/146431771-2ef99427-525a-49d8-bc18-aee990c40af8.png)

Could you check on your end please. I've added the image being used to the wpcom diff.

thedebian committed:
[not verified] Downsize VideoPress example image and only use an img block as example

thedebian commented:
> Could you check on your end please. I've added the image being used to the wpcom diff.

I still can't reproduce this, but I went ahead and reduced the image size + cleaned up the output a bit by only returning an img (as we don't care about the caption anymore);
Is it better now ?
It's supposed to be max-width 100% too, from what I can see in the inspector, but I can force it somehow on the block output itself or something along the line if it doesn't get any better.

samiff commented:
@thedebian I figured out what was different on my end - I'm running WP 5.9 nightlies and that along with a theme like Twenty Nineteen seems to change the editor wrapper styles being applied, causing the scrollbars I've seen. I'm thinking this one should be good to go 👍

thedebian commented:
@samiff Sorry, I tried resyncing the wpcom PR by rebasing this diff, so the ✅ is now gone here (and I had to manually sync the wpcom diff anyway because of the new image file 🙄 ); Could you reapprove it please so it could be merged ?

thedebian commented:
Changeset 237530
